osm2gtfs/core/osm_connector.py

```diff
@@ -222,116 +242,118 @@ def get_stops(self, refresh=False):
 
         # Obtain raw data about routes from OpenStreetMap
         result = self._query_stops()
+        self.stops['regular'] = {}
+        self.stops['stations'] = {}
 
         # Build stops from ways (polygons)
         for stop in result.ways:
-            if self._is_valid_stop_candidate(stop):
-                self.stops["way/" + str(stop.id)
-                           ] = self._build_stop(stop, "way")
+            osm_type = "way"
+            stop_object = self._build_stop(stop, osm_type)
+            if stop_object:
+                self.stops['regular'][osm_type + "/" + str(
+                    stop_object.osm_id)] = stop_object
 
         # Build stops from nodes
         for stop in result.nodes:
-            if self._is_valid_stop_candidate(stop):
-                self.stops["node/" + str(stop.id)
-                           ] = self._build_stop(stop, "node")
-
-        # Build stop_areas
-        for relation in result.relations:
-            # valid stop_area candidade?
-            if 'public_transport' in relation.tags:
-                if relation.tags["public_transport"] == "stop_area":
-                    self.stops["relation/" + str(relation.id)
-                               ] = self._build_stop_area(relation)
+            osm_type = "node"
+            stop_object = self._build_stop(stop, osm_type)
+            if stop_object:
+                self.stops['regular'][osm_type + "/" + str(
+                    stop_object.osm_id)] = stop_object
+
+        # Build stations from stop_area relations
+        for stop in result.relations:
+            osm_type = "relation"
+            stop_object = self._build_station(stop, osm_type)
+            if stop_object:
+                self.stops['stations'][osm_type + "/" + str(
+                    stop.id)] = stop_object
```

Comment:
I had this back and forth and I liked this style more for this particular piece.

It is actually not doing the same thing, each of the lines are slightly different (they loop through different lists, they define different strings for `osm_type`, they call eventually different functions to build a stop or a station. It is correct that this is not too elegant, but the best I could come up with. Outfactoring made it less readable and longer. Please try to give a concrete proposal, in case you want to optimize it - I guess you will see once you start doing.